ilc/server/tailor/merge-configs.spec.ts

```diff
@@ -390,14 +417,97 @@ describe('merge configs', () => {
                                         firstPropOfConstSlotSecondProp: 'firstPropOfConstSlotSecondProp',
                                     },
                                 },
+                                kind: null,
                             },
                             new: {
                                 appName: apps['@portal/const'].name,
                             },
                         },
+                        versionId: 'v1.0.5',
                     },
                 ],
             });
         });
+        it('should override route only with same domains (no domain)', () => {
+            const overrideConfig = {
+                routes: [
+                    {
+                        routeId: 5,
+                        route: '*',
+                        next: false,
+                        slots: {},
+                        meta: {}, // Added meta
+                        versionId: 'v1.0.5', // Added versionId
+                        domainId: 7,
+                    },
+                ],
+            };
+
+            expect(mergeConfigs(registryConfig, overrideConfig)).to.be.eql(registryConfig);
+        });
+        it('should override route only with same domains (no domain) (merge)', () => {
+            const overrideConfig = {
+                routes: [
+                    {
+                        routeId: 1,
+                        route: '*',
+                        next: false,
+                        slots: {},
+                        meta: {}, // Added meta
+                        versionId: 'v1.0.5', // Added versionId
+                        domainId: 7,
+                    },
+                ],
+            };
+
+            expect(mergeConfigs(registryConfig, overrideConfig)).to.be.eql(registryConfig);
+        });
+        it('should override route only with same domains (domain present)', () => {
+            const overrideConfig = {
+                routes: [
+                    {
+                        routeId: 5,
+                        route: '/anotherdomain',
+                        next: false,
+                        orderPos: 999,
+                        slots: {},
+                        meta: {}, // Added meta
+                        versionId: 'v1.0.5', // Added versionId
+                        domainId: 7,
+                    },
+                ],
+            };
+
+            const mergedConfig = {
```

Comment:
Can we not perform any actual logic that can change the result of merging? I think merging function should check explicitly objects that we are awaiting. Destructurization here is a part of the logic itself which may lead to not something we are expecting. 

Let's do dumb check here